### PR TITLE
Refactor data source to throw exception instead of returning optionals

### LINKED ISSUE #707: Refactor data sources to throw exceptions instead of returning Optionals
Currently, if reading grid, participant components etc., an empty Optional is returned, accompanied by some error log messages that are writting into console, but which might be missed.

Refactor methods such as `CsvSystemParticipantSource#getSystemParticipants`, `CsvRawGridSource#getGridData` to no longer return Optionals, but throw appropriate exceptions.

### PR DESCRIPTION
**Pull request by staudtMarius**
Resolves #707

### CODE REVIEW

**Review by danielfeismann — CHANGES_REQUESTED** [review]
Thanks for your work done here! This looks all fine for me, just a minor question.

**Review by danielfeismann — COMMENTED** [review]
LGTM and could merged in my point of view. @sebastian-peter mentioned that he also wanted to have a look here

**Comment by staudtMarius** [comment]
!test

**Review by sebastian-peter — CHANGES_REQUESTED** [review]
Looks good overall, your coding is improving! 

Could you take care of these points, before I re-review? 
In general, please also have an eye on SonarQube code smells.

**Comment by staudtMarius** [comment]
!test

**Comment by staudtMarius** [comment]
test

**Comment by staudtMarius** [comment]
!test

**Review by sebastian-peter — CHANGES_REQUESTED** [review]
Some more comments from my side :)

File: src/main/java/edu/ie3/datamodel/io/source/ThermalSource.java [hunk 1]
```diff
@@ -235,71 +241,72 @@ protected Stream<Optional<ThermalUnitInputEntityData>> buildThermalUnitInputEnti
...
-      ThermalHouseInputFactory factory) {
-    return assetInputEntityDataStream(ThermalHouseInput.class, typeSource.getOperators())
-        .flatMap(
-            assetInputEntityData ->
-                buildThermalUnitInputEntityData(assetInputEntityData, getThermalBuses())
-                    .filter(Optional::isPresent)
-                    .map(Optional::get)
-                    .map(factory::get))
-        .collect(Collectors.toSet());
+  public Try<Set<ThermalHouseInput>, FailureException> buildThermalHouseInputEntities(
```
[the remark this answers is not in the document]
Comment (staudtMarius) [reply src/main/java/edu/ie3/datamodel/io/source/ThermalSource.java:262]: Thanks. I missed that part.

**Review by danielfeismann — APPROVED** [review]

**Pull request merged**